### DISCUSSION TITLE
Multiple rq-workers, queue priority

### DISCUSSION THREAD

**Discussion by julianstolp** [netbox-community/netbox#9626] [Q&A]
We have a cronjob which queues a script to scan a switch. It is done for all of our switches every night. It takes around 6 hours to complete with one worker. In between these 6 hours all manual started scripts by a user are queued behind and are not executed directly. So I searched for a way to spawn more workers and set a low priority to the scan.

**Worker**
I tried as described in [#5927](https://github.com/netbox-community/netbox/issues/5927) (Renaming netbox-rq.service to netbox-rq@.service, perform a  systemctl daemon-reload, start multiple workers with a number behind @) which seems works fine.
![image](https://user-images.githubusercontent.com/41152397/176419941-2468ca68-5280-4866-94ac-4a0c3179179f.png)

Are there any possible sideeffects like datacorruption or so?

**Queue**
In the issue [#6651](https://github.com/netbox-community/netbox/issues/6651) it's stated that it is possible for plugins to access high or low priority queue. Is this also possible for custom scripts to be queued in high?

Thanks!

**Comment by julianstolp** [comment]
Does anyone have an answer to this topic?

**Comment by mtinberg** [comment]
My understanding is that the workers pick up their jobs from a queue in redis so it's specifically designed for there to be multiple possible workers and only one will pick up the queued job.  I have workers on different servers and that has been fine (as long as they both have the same code for Scripts and Reports)


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
